gateway-types/src/ignition/v2.rs

```diff
@@ -0,0 +1,151 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use crate::component::SpIdentifier;
+
+#[derive(Deserialize, JsonSchema)]
+pub struct PathSpIgnitionCommand {
+    /// ID for the SP that the gateway service translates into the appropriate
+    /// port for communicating with the given SP.
+    #[serde(flatten)]
+    pub sp: SpIdentifier,
+    /// Ignition command to perform on the targeted SP.
+    pub command: IgnitionCommand,
+}
+
+#[derive(
+    Debug,
+    Clone,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Deserialize,
+    Serialize,
+    JsonSchema,
+)]
+pub struct SpIgnitionInfo {
+    pub id: SpIdentifier,
+    pub details: SpIgnition,
+}
+
+/// Ignition command.
+#[derive(
+    Debug, Clone, Copy, PartialEq, Eq, Deserialize, Serialize, JsonSchema,
+)]
+#[serde(rename_all = "snake_case")]
+pub enum IgnitionCommand {
+    PowerOn,
+    PowerOff,
+    PowerReset,
+}
+
+impl From<IgnitionCommand> for gateway_messages::IgnitionCommand {
+    fn from(cmd: IgnitionCommand) -> Self {
+        match cmd {
+            IgnitionCommand::PowerOn => {
+                gateway_messages::IgnitionCommand::PowerOn
+            }
+            IgnitionCommand::PowerOff => {
+                gateway_messages::IgnitionCommand::PowerOff
+            }
+            IgnitionCommand::PowerReset => {
+                gateway_messages::IgnitionCommand::PowerReset
+            }
+        }
+    }
+}
+
+/// State of an ignition target.
+///
+/// TODO: Ignition returns much more information than we're reporting here: do
+/// we want to expand this?
+#[derive(
+    Debug,
+    Clone,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Deserialize,
+    Serialize,
+    JsonSchema,
+)]
+#[serde(tag = "present")]
+pub enum SpIgnition {
+    #[serde(rename = "no")]
+    Absent,
+    #[serde(rename = "yes")]
+    Present {
+        id: SpIgnitionSystemType,
+        power: bool,
+        ctrl_detect_0: bool,
+        ctrl_detect_1: bool,
+        flt_a3: bool,
+        flt_a2: bool,
+        flt_rot: bool,
+        flt_sp: bool,
```

Comment:
Not sure if this makes sense or not, please ignore this comment if it doesn't. It could be nice to have the fields be more descriptive. For those of us without a background in firmware eng they aren't super obvious. How about something like `faults_power_a3`, `controller_detect_0` etc?

Reply:
Changing field names is pretty intrusive, I'll see about adding some more comments